keys/public.go

```diff
@@ -0,0 +1,151 @@
+package keys
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"io/ioutil"
+
+	"crypto/cipher"
+
+	"crypto/aes"
+
+	"encoding/json"
+
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// PublicKey is the public key part of the KeyPair
+type PublicKey struct {
+	key       *rsa.PublicKey
+	encrypter Encrypter
+}
+
+// AsBytes returns the key as pem formatted byte array
+func (pk *PublicKey) AsBytes() ([]byte, error) {
+	der, err := x509.MarshalPKIXPublicKey(pk.key)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal public key")
+	}
+
+	var pemkey = &pem.Block{
+		Type:  "PUBLIC KEY",
+		Bytes: der,
+	}
+
+	return pem.EncodeToMemory(pemkey), nil
+}
+
+// AsString returns the key as pem formatted string
+func (pk *PublicKey) AsString() (string, error) {
+	bytes, err := pk.AsBytes()
+	if err != nil {
+		return "", err
+	}
+	return string(bytes), err
+}
+
+// ToFile writes the key to disk in pem format
+func (pk *PublicKey) ToFile(path string) error {
+	bytes, err := pk.AsBytes()
+	if err != nil {
+		return err
+	}
+
+	// TODO check permissions
+	err = ioutil.WriteFile(path, bytes, 0744)
+	if err != nil {
+		return fmt.Errorf("failed to write public key to file %s: %w", path, err)
+	}
+	return nil
+}
+
+// Encrypt encrypts the given input
+// In cases where the input can not be encrypted with RSA because it is too long,
+// we switch to a hybrid encryption (i. e. using symmetric crypto for the actual content via a randomly generated key which in turn is encrypted with RSA)
```

Comment:
```suggestion
// Encrypt encrypts the given input.
// In cases where the input can not be encrypted with RSA because it is too long,
// we switch to a hybrid encryption (i. e. using symmetric crypto for the actual content via a randomly generated key which in turn is encrypted with RSA).
```